server/melinda-io-service.js

```diff
@@ -25,138 +25,136 @@
+    return Promise.reject(new RecordIOError(HttpStatus.BAD_REQUEST, errorMessage));
   }
 
-  return updateRecord(client, record).then(function(updateResponse) {
+  return updateRecord(client, record).then(updateResponse => {
     logger.log('info', `Record updated ok for ${recordId}`, updateResponse.messages);
 
-    const recordIdFromUpdate = updateResponse.recordId;
-
-    return loadRecord(client, recordIdFromUpdate, { include_parent: 1}).then(record => {
+    return loadRecord(client, recordId).then((record) => {
       return record;
     }).catch(error => {
-      logger.log('info', `Error loading record ${recordId}`, error);
+      logger.log('error', `@updateAndReloadRecord -> Error loading record ${recordId}\n${JSON.stringify(error)}`);
       throw error;
     });
-
   }).catch(error => {
-    logger.log('info', `Record update failed for ${recordId}`, error);
+    logger.log('info', `Record update failed for ${recordId} `, error);
     if (looksLikeMelindaClientParseError(error)) {
-
       const reason = _.get(error, 'errors[0].message').substr('melinda-api-client unable to parse: '.length);
-
-      throw new RecordIOError(reason, HttpStatus.BAD_REQUEST);
+      throw new RecordIOError(HttpStatus.BAD_REQUEST, reason);
     }
     throw error;
   });
 }
 
 export function createAndReloadRecord(client, record) {
-
-  return createRecord(client, record).then(function(updateResponse) {
+  return createRecord(client, record).then(function (updateResponse) {
     const recordId = updateResponse.recordId;
 
-    logger.log('info', `Record created, id: ${recordId}`, updateResponse.messages);
+    logger.log('info', `Record created, id: ${recordId} `, updateResponse.messages);
+
+    return loadRecord(client, recordId).then(result => {
+      return _.assign({}, result, {recordId});
 
-    return loadRecord(client, recordId, { include_parent: 1}).then(result => {
-      return _.assign({}, result, { recordId });
-      
     }).catch(error => {
-      logger.log('info', `Error loading record ${recordId}`, error);
+      logger.log('error', `@createAndReloadRecord -> Error loading record ${recordId}\n${JSON.stringify(error)}`);
       throw error;
     });
 
   }).catch(error => {
     logger.log('info', 'Record creation failed', error);
     if (looksLikeMelindaClientParseError(error)) {
-
       const reason = _.get(error, 'errors[0].message').substr('melinda-api-client unable to parse: '.length);
 
-      throw new RecordIOError(reason, HttpStatus.BAD_REQUEST);
+      throw new RecordIOError(HttpStatus.BAD_REQUEST, reason);
     }
     throw error;
   });
 }
 
-
-
-
 function looksLikeMelindaClientParseError(error) {
-  return _.get(error, 'errors[0].code') === -1 && _.get(error, 'errors[0].message','').startsWith('melinda-api-client unable to parse: ');
+  return _.get(error, 'errors[0].code') === -1 && _.get(error, 'errors[0].message', '').startsWith('melinda-api-client unable to parse: ');
```

Comment:
If we're not using melinda-api-client, we'll probably not get these types of errors anymore?